include/dlaf/eigensolver/backtransformation/mc.h

```diff
@@ -0,0 +1,227 @@
+}
+
+template <Device device, class T>
+void copySingleTile(hpx::shared_future<matrix::Tile<const T, device>> in,
+                    hpx::future<matrix::Tile<T, device>> out) {
+  hpx::dataflow(dlaf::getCopyExecutor<Device::CPU, Device::CPU>(),
+                matrix::unwrapExtendTiles(matrix::copy_o), in, out);
+}
+
+template <class Executor, Device device, class T>
+void trmmPanel(Executor&& ex, hpx::shared_future<matrix::Tile<const T, device>> t,
+               hpx::future<matrix::Tile<T, device>> w) {
+  hpx::dataflow(ex, matrix::unwrapExtendTiles(tile::trmm_o), blas::Side::Right, blas::Uplo::Upper,
+                blas::Op::ConjTrans, blas::Diag::NonUnit, T(1.0), t, w);
+}
+
+template <class Executor, Device device, class T>
+void gemmUpdateW2(Executor&& ex, hpx::future<matrix::Tile<T, device>> w,
+                  hpx::shared_future<matrix::Tile<const T, device>> c,
+                  hpx::future<matrix::Tile<T, device>> w2) {
+  hpx::dataflow(ex, matrix::unwrapExtendTiles(tile::gemm_o), blas::Op::ConjTrans, blas::Op::NoTrans,
+                T(1.0), w, c, T(1.0), std::move(w2));
+}
+
+template <class Executor, Device device, class T>
+void gemmTrailingMatrix(Executor&& ex, hpx::shared_future<matrix::Tile<const T, device>> v,
+                        hpx::shared_future<matrix::Tile<const T, device>> w2,
+                        hpx::future<matrix::Tile<T, device>> c) {
+  hpx::dataflow(ex, matrix::unwrapExtendTiles(tile::gemm_o), blas::Op::NoTrans, blas::Op::NoTrans,
+                T(-1.0), v, w2, T(1.0), std::move(c));
+}
+
+// Implementation based on:
+// 1. Part of algorithm 6 "LAPACK Algorithm for the eigenvector back-transformation", page 15, PhD thesis
+// "GPU Accelerated Implementations of a Generalized Eigenvalue Solver for Hermitian Matrices with
+// Systematic Energy and Time to Solution Analysis" presented by Raffaele Solcà (2016)
+// 2. Report "Gep + back-transformation", Alberto Invernizzi (2020)
+// 3. Report "Reduction to band + back-transformation", Raffaele Solcà (2020)
+// 4. G. H. Golub and C. F. Van Loan, Matrix Computations, chapter 5, The Johns Hopkins University Press
+template <class T>
+struct BackTransformation<Backend::MC, Device::CPU, T> {
+  static void call_FC(Matrix<T, Device::CPU>& mat_c, Matrix<const T, Device::CPU>& mat_v,
+                      common::internal::vector<hpx::shared_future<common::internal::vector<T>>> taus);
+  static void call_FC(comm::CommunicatorGrid grid, Matrix<T, Device::CPU>& mat_c,
+                      Matrix<const T, Device::CPU>& mat_v,
+                      common::internal::vector<hpx::shared_future<common::internal::vector<T>>> taus);
+};
+
+template <class T>
+void BackTransformation<Backend::MC, Device::CPU, T>::call_FC(
+    Matrix<T, Device::CPU>& mat_c, Matrix<const T, Device::CPU>& mat_v,
+    common::internal::vector<hpx::shared_future<common::internal::vector<T>>> taus) {
+  using hpx::util::unwrapping;

```

Comment:
```suggestion
  using hpx::unwrapping;

```
You should also receive a warning if you rebase on master now.